net: Fix CThreadInterrupt::sleep_for TSan issues

hebasto opened the pull request:
This PR adds to the `ThreadDNSAddressSeed` and `ThreadOpenAddedConnections` threads their own `CThreadInterrupt` objects, and fixes ThreadSanitizer's "double lock of a mutex" warnings.

Fix #19024

hebasto committed:
net: Add m_interrupt_dnsseed_thread

hebasto committed:
net: Add m_interrupt_addcon_thread

MarcoFalke commented on src/net.h:
```diff
@@ -460,6 +460,8 @@ class CConnman
     std::atomic<bool> flagInterruptMsgProc{false};
 
+    CThreadInterrupt g_interrupt_dnsseed_thread;
+    CThreadInterrupt g_interrupt_addcon_thread;
```
```suggestion
    CThreadInterrupt m_interrupt_addcon_thread;
```

Looks like a member to me, no?

hebasto replied:
Sure! Thank you. [Updated](https://github.com/bitcoin/bitcoin/pull/19029#issuecomment-631595378).

hebasto commented:
Updated 7bb5707a533090e2b1cdc7e27b860b37b25651e1 -> a46122ca6503c780b303d2c8194625faeb8d0f65 ([pr19029.01](https://github.com/hebasto/bitcoin/commits/pr19029.01) -> [pr19029.02](https://github.com/hebasto/bitcoin/commits/pr19029.02), [diff](https://github.com/hebasto/bitcoin/compare/pr19029.01..pr19029.02)):

- addressed @MarcoFalke's [comment](https://github.com/bitcoin/bitcoin/pull/19029#discussion_r428153408):

> Looks like a member to me, no?

hebasto commented:
I'm not sure these warnings aren't false positive due to the Fedora clang's bug. See: https://github.com/bitcoin/bitcoin/issues/19024#issuecomment-631935873

So closing for now.

MarcoFalke commented:
An alternative would be to make the mutex recursive for now

hebasto commented:
> An alternative would be to make the mutex recursive for now

This won't compile, unfortunately.

hebasto commented:
Rebased a46122ca6503c780b303d2c8194625faeb8d0f65 -> bad0f26b6c73678c4282343978f75a84b537923a ([pr19029.02](https://github.com/hebasto/bitcoin/commits/pr19029.02) -> [pr19029.03](https://github.com/hebasto/bitcoin/commits/pr19029.03)) due to the conflict with #16939.

hebasto commented:
Closing for now. See: https://github.com/bitcoin/bitcoin/issues/19024#issuecomment-636479764
